src/main/scala/com/microsoft/hyperspace/index/FileBasedSignatureProvider.scala

```diff
@@ -57,7 +63,9 @@ class FileBasedSignatureProvider extends LogicalPlanSignatureProvider {
           _) =>
         fingerprint ++= location.allFiles.foldLeft("")(
```

Comment:
The ++= seems from a legacy code (sorry about that). Could you please replace it with += which is more relevant for String concatenation?
Reason: String also extends `TraversableLike` in the sense that it wraps over `Seq[Char]` and _technically_ ++= is not incorrect, only semantically it makes more sense to concatenate strings as +=.
Again, looks like it's from past code but if you can do it in this PR it would be helpful.

Alternately, StringBuilder is a better option for appending strings

Reply:
if not, it's ok to leave it as it is. We can also come back to it in future.